docs/development/pigs/pig-010.rst

```diff
@@ -0,0 +1,460 @@
+replaced by documentation examples, or improved to in some cases used regions.
+This is not high priority and a separate task, since this functionality isn't
+used anywhere else in Gammapy.
+
+Outlook
+=======
+
+This PIG only describes some short-term improvements to spatial regions.
+Long-term, we might want to develop a more general solution for regions that
+also includes non-spatial regions, e.g. spectral regions or time regions.
+
+Just to mention what we have at the moment: spectral "regions" are handled via
+``energy_bounds`` or ``(energy_min, energy_max)`` in a not very consistent way.
+We could look towards the Astropy `specutils`_ package and the
+`specutils.SpectralRegion`_ object as an example how to implement spectral
+regions. Currently there is plan to integrate ``astropy-regions`` and
+``specutils``, `astropy-regions GH 228`_ has been opened for discussion.
+
+Supporting an energy-dependent maximum field of view radius selection, or a
+field-of-view dependent safe energy offset might be a motivation to couple
+spatial and spectral region, possibly supported via ``gammapy.maps``.
+
+For time regions we currently often have ``(time_min, time_max)`` in the API,
+and we have the `gammapy.data.GTI`_ class to represent multiple intervals. It's
+not clear if this needs to be changed or wrapped somehow into a "time region",
+or if there are any use cases to couple time with the spatial and spectral
+regions, e.g. for a time-dependent safe energy threshold or user analysis
+option.
+
+These thoughts are related to the question in `GH 1805`_: "Introduce RegionGeom
+and RegionNDMap?" - Axel: do you want this for v1.0 or postpone? If v1.0, then
+do you want to flesh out a proposal here and now, or later in a separate PIG?
+
+In the HESS software, there is a region class that wraps a mask, and that is
+sometimes useful, specifically because in the HESS software exclusion regions
+are region objects, sometimes compound with a mask region and other regions like
+circles. It's clearly an option, easy to implement as a ``SkyRegion`` sub-class.
+But also causes problems, especially when it comes to serialisation. Thoughts?
+I'm OK to not add this for now, and just use ``Map`` objects for exclusion
+masks, and require that users create that before using Gammapy, in case they
+e.g. have a list of circles or whatever they want to use as exclusion region.
+
+Task list
+=========
+
+We propose to implement this PIG though a series of small pull requests. Some are
+dependent on others, but for many also the order doesn't matter and overall the
+amount of work is limited.
+
+- Add ``PixRegion.rotate(center, angle)`` in ``astropy-healpix``. See `Rotated regions`_ above.
+- Release ``astropy-healpix`` v0.4, then bump Gammapy dependency to that version.
+- Improve rotated region code and tests and docs in Gammapy.
+  This could mean a rebase and continuation of `GH 2089`_, or closing that and
```

Comment:
I would this will mean adapting was is implemented in #2089.

Note that the addition of `PixRegion.rotate(angle)` only marginally simplify the code. A central problem is to have a determination of the apparent angular size a seen from the center of rotation. e.g. something like `arctan(2*radius/offset)` for a circular region. 
An interesting addition would be to have some method or property giving the maximum extent in degrees/pixels  for any region. Maybe this can be done through `Region.to_polygon()` and then compute the distance between every pair of point. This is not very nice though. At the moment, the code uses the `bounding_box` of the `PixRegion` but this is not a very satisfactory solution (and not all regions have a `bounding_box`).

Reply:
@registerrier - Is this really needed? For a bit of efficiency to have fewer test positions for off regions? Why not just start rotating at the actual position by small angles and have simpler code at the cost of a bit of computation that could be avoided?

You have to decide if you want to finish and merge in #2089 or abandon it. If you want it in, please make it a priority so that others aren't blocked to work on that regions-related code. If you think re-starting from master is more efficient, let's do that.

Could you comment once more, and I'll update the PIG to reflect that decision.

Reply:
@registerrier - I have added this to the proposal:

> One complication is that in the current code, to have fewer off test regions,
the algorithm takes a large step ``arctan(2*radius/offset)`` at the start and
after placing an off region. This requires a "radius", which is is not available
for regions of a complex shape. We propose that this optimisation is only left
for the circle case (which already has a specialised fast algorithm anyways
using a distance mask), and for other regions for now the normal thing is done
to test all positions in small angle increments.